react-mockup/src/components/app/App.js

```diff
@@ -3,14 +3,19 @@ import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
 import Home from '../Home';
 import HumFeed from '../hums/HumFeed';
 import UserDetail from '../user/UserDetail';
+import Callback from '../../containers/auth/Callback';
+import { withSession } from '../../containers/auth/withSession';
+import Loading from './Loading';
 
 function App() {
   return (
     <Router>
       <Switch>
-        <Route exact path="/" component={Home} />
-        <Route exact path="/hum-feed" component={HumFeed} />
-        <Route exact path="/user" component={UserDetail} />
+        <Route exact path="/" component={withSession(Home)} />
```

Comment:
```suggestion
        <Route path="/hum-feed" component={withSession(HumFeed)} />
        <Route path="/user" component={withSession(UserDetail)} />
        <Route path="/loading" component={Loading} />
        <Route path="/callback" component={Callback} />
        <Route exact path="/" component={withSession(Home)} />
```
You should be able to arrange your routes in such a way that exact path is only needed on the default '/' path when using a Switch. Just make sure you list the most specific routes first, ie /hum/:id before /hum.